threeML/plugins/OGIP/likelihood_functions.py

```diff
@@ -114,4 +114,111 @@ def poisson_observed_gaussian_background(observed_counts, background_counts, bac
     log_likes[nidx] = observed_counts[nidx] * np.log(expected_model_counts[nidx] + 1e-100) - \
                       expected_model_counts[nidx] - logfactorial(observed_counts[nidx])
 
-    return np.sum(log_likes)
\ No newline at end of file
+    return np.sum(log_likes)
+
+
+# For some statistical functions (WAIC) it is desirable to look at the likelihood of each individual data point.
+# These functions return the vector of the observation for that purpose
+
+def poisson_log_likelihood_ideal_bkg_vector(observed_counts, expected_bkg_counts, expected_model_counts):
```

Comment:
ugh! :-)

Is this a copy and paste? I'd rather not do this, as possible errors in the original function will need to be corrected in two places...

Instead, let's have the old functions always return the vector of log likes, and the OGIPLike plugin perform a np.sum().

I can take care of this.